django/template/base.py

```diff
@@ -187,12 +187,13 @@ def __init__(self, template_string, origin=None, name=None, engine=None):
         if engine is None:
             from .engine import Engine
             engine = Engine.get_default()
-        if engine.debug and origin is None:
+        if origin is None:
             origin = StringOrigin(template_string)
-        self.nodelist = engine.compile_string(template_string, origin)
         self.name = name
         self.origin = origin
         self.engine = engine
+        self.source = template_string
+        self.nodelist = engine.compile_string(self)
```

Comment:
Good point. It may make more sense to move these methods to ``Template`` as ``Template.compile_nodelist`` and ``Template.get_exception_info``. These methods aren't really used by the engine itself.